tests/Agent/IntegrationTests/UnboundedIntegrationTests/Elasticsearch/ElasticsearchAsyncTests.cs

```diff
@@ -0,0 +1,280 @@
+    {
+        const string SYNC_MODE = "async";
+
+        protected ElasticsearchAsyncTestsBase(TFixture fixture, ITestOutputHelper output, string clientType) : base(fixture, output, clientType, SYNC_MODE)
+        {
+
+        }
+
+        [Fact]
+        public void IndexAndSearch()
+        {
+            var expectedTransactionName = "OtherTransaction/Custom/MultiFunctionApplicationHelpers.NetStandardLibraries.Elasticsearch.ElasticsearchExerciser/Index";
+
+            var expectedMetrics = new List<Assertions.ExpectedMetric>
+            {
+                new Assertions.ExpectedMetric { metricName = $"statement/Elasticsearch/flights/Index", metricScope = expectedTransactionName, callCount = 1 },
+            };
+
+            var metrics = _fixture.AgentLog.GetMetrics().ToList();
+
+            var spanEvents = _fixture.AgentLog.GetSpanEvents();
+
+            var traceId = spanEvents.Where(@event => @event.IntrinsicAttributes["name"].ToString().Equals(expectedTransactionName)).FirstOrDefault().IntrinsicAttributes["traceId"];
+
+            var operationDatastoreSpans = spanEvents.Where(@event => @event.IntrinsicAttributes["traceId"].ToString().Equals(traceId) && @event.IntrinsicAttributes["name"].ToString().Contains("Datastore/statement/Elasticsearch"));
+
+
+            NrAssert.Multiple
+            (
+                () => Assertions.MetricsExist(expectedMetrics, metrics),
+                () => Assert.Equal(1, operationDatastoreSpans.Count())
+            );
+        }
+
+    }
+
+    #region NEST
+    [NetFrameworkTest]
+    public class ElasticsearchNestAsyncTestsFWLatest : ElasticsearchAsyncTestsBase<ConsoleDynamicMethodFixtureFWLatest>
+    {
+        public ElasticsearchNestAsyncTestsFWLatest(ConsoleDynamicMethodFixtureFWLatest fixture, ITestOutputHelper output)
+            : base(fixture, output, ClientType.NEST.ToString())
+        {
+        }
+    }
+
+    [NetFrameworkTest]
+    public class ElasticsearchNestAsyncTestsFW48 : ElasticsearchAsyncTestsBase<ConsoleDynamicMethodFixtureFW48>
+    {
+        public ElasticsearchNestAsyncTestsFW48(ConsoleDynamicMethodFixtureFW48 fixture, ITestOutputHelper output)
+            : base(fixture, output, ClientType.NEST.ToString())
+        {
+        }
+    }
+
+    [NetFrameworkTest]
+    public class ElasticsearchNestAsyncTestsFW471 : ElasticsearchAsyncTestsBase<ConsoleDynamicMethodFixtureFW471>
+    {
+        public ElasticsearchNestAsyncTestsFW471(ConsoleDynamicMethodFixtureFW471 fixture, ITestOutputHelper output)
+            : base(fixture, output, ClientType.NEST.ToString())
+        {
+        }
+    }
+
+    [NetFrameworkTest]
+    public class ElasticsearchNestAsyncTestsFW462 : ElasticsearchAsyncTestsBase<ConsoleDynamicMethodFixtureFW462>
+    {
+        public ElasticsearchNestAsyncTestsFW462(ConsoleDynamicMethodFixtureFW462 fixture, ITestOutputHelper output)
+            // FW462 is testing MongoDB.Driver version 2.3, which needs to connect to the 3.2 server
```

Comment:
This comment might have been accidentally included.

Reply:
Yeah, lot of copy-paste :)